rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java

```diff
@@ -765,6 +761,8 @@ protected void shutdownNodeSourceServices(Client initiator) {
 
         this.activePolicy.shutdown(initiator);
         this.infrastructureManager.internalShutDown();
+        this.rmcore.disconnect(Client.getId(PAActiveObject.getStubOnThis()));
```

Comment:
maybe it's better to do this inside finishNodeSourceShutdown ?

Reply:
Yes it makes more sense. Applied.